### PR TITLE
Safari bugs - Page not loading, Column Height

### PR DESCRIPTION
**Pull request by stephenburgess8**
Tested in Firefox, Chrome, and Safari on Mac. #2899 

- Fix bug of rightmost column to extend beyond the bottom of the frame in Safari.
- Adjust padding on the column so scrolling looks like other columns.
- Use dynamic classnames rather than dynamically added styles for the mastodon column.
- Do NOT use inline styles.
- Fix column making whole page scroll by using display: none instead of visibility: hidden.

Screenshots:

Before and After in Chrome. The difference: See how the rightmost column extends below the other columns and there is no scrollbar? Now, in the after picture there IS a scrollbar and it extends the same height as the other columns
screen shot 2017-05-13 at 12 55 03 pmscreen shot 2017-05-13 at 12 59 51 pm

Before and After Safari:
screen shot 2017-05-13 at 1 11 26 pmscreen shot 2017-05-13 at 1 14 37 pm

### CODE REVIEW

**Review by ykzts — APPROVED** [review]

**Comment by stephenburgess8** [comment]
@akihikodaki Any reason why these styles would be needed that is not immediately obvious? Thanks.

**Comment by akihikodaki** [comment]
@stephenburgess8 I commented to the issue. https://github.com/tootsuite/mastodon/issues/2899#issuecomment-299890927

**Review by akihikodaki — CHANGES_REQUESTED** [review]
As I mentioned at https://github.com/tootsuite/mastodon/issues/2899#issuecomment-299890927.

**Review by akihikodaki — CHANGES_REQUESTED** [review]
Still failing; open a timeline in multi-column layout and then make the screen smaller.

**Comment by akihikodaki** [comment]
Thank you for working on this. I do not have Mac, so that helps much.

**Comment by stephenburgess8** [comment]
I have been having to recompile with webpack in order to get my CSS changes to show up. After I make this change and recompile, this change shows up and works for me. Can you confirm that `overflow-y` is compiled as `auto` and not `scroll` on the `scrollable` class? That said, I would like to spend a little time with this and do some more cross-browser testing before accepting it. `scrollable` class is used in many places.

**Comment by akihikodaki** [comment]
Ah, sorry. Actually webpack was failing as you pointed out. However I now see it goes wrong on Chrome with the single-column layout.

**Comment by stephenburgess8** [comment]
I can return to this a little later today.

File: app/javascript/styles/containers.scss [hunk 1]
```diff
@@ -9,11 +9,29 @@
...
+  }
+}
+
 .mastodon-column-container {
   display: flex;
   flex: 1 1 auto;
 }
 
+.mastodon-column-container--hidden {
+  display: none;
```
Comment (akihikodaki) [inline app/javascript/styles/containers.scss]: This breaks scroll (and possibly other states) preservation.

**Review by akihikodaki — CHANGES_REQUESTED** [review]
This breaks scroll (and possibly other states) preservation.
Also, lease rebase rather than creating a merge commit instead.

**Comment by akihikodaki** [comment]
Perhaps it is too difficult to keep everything correct on every supported browsers.
The current design of the UI is inconsistent in terms that only the scroll bar is tied to the window, leaving the column independent of the window in the single-column layout. In the multiple-column layout, each column has the scroll bar and they are completely independent of the window.
We may need issues or pull requests to fix the inconsistency.

File: app/javascript/styles/containers.scss (hunk 1 above)
Comment (akihikodaki) [inline app/javascript/styles/containers.scss]: (reproducible on `Mozilla/5.0 (X11; Linux x86_64; rv:53.0) Gecko/20100101 Firefox/53.0`, with the single-column layout)

**Comment by stephenburgess8** [comment]
I'm sorry, I hadn't registered that you were doing visibility:hidden in order to save the scroll position.

When you get a chance, if you could please check this out one more time. I have made it so that scroll position is saved. I have tested in the following browsers in both desktop view and smaller mobile view:

Safari 9
Firefox 53.0.2 
Chrome 58.0.3029.110

NOT IE11 (have not been able to get my VM to work with localhost:3000)

There is a small visual bug in Safari in Mobile view but it is still usable and it's a much better Safari experience overall now. Should have a slightly improved UI in Chrome and Firefox.

**Comment by akihikodaki** [comment]
The composer died on `Mozilla/5.0 (X11; Linux x86_64; rv:53.0) Gecko/20100101 Firefox/53.0`, with the mobile view. :/

**Comment by akihikodaki** [comment]
By the way, Mastodon does not work on IE. Test with Edge for Windows.

**Comment by stephenburgess8** [comment]
The wrapper I added around mobile column must be disrupting the compose. However, I think it's getting closer. I think it's worth it to provide support for more browsers, as that makes the software accessible to more people. Thanks for your help on this.

**Comment by akihikodaki** [comment]
b94f0bd5db54fdb7487f145865bbea435d01fdbc looks good for me (`Mozilla/5.0 (X11; Linux x86_64; rv:53.0) Gecko/20100101 Firefox/53.0` and `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36`)

**Comment by stephenburgess8** [comment]
Sorry, display none was not needed. Rechecked in Safari, Chrome, Firefox

**Review by ykzts — APPROVED** [review]

**Comment by ykzts** [comment]
@stephenburgess8 Can you solve this problem in #3062?

**Comment by Gargron** [comment]
I think this is no longer a problem.

**Pull request closed without merging**